### DISCUSSION TITLE
Unavailable Sensors

### DISCUSSION THREAD

**Discussion by imcintyr** [PeteRager/lennoxs30#305] [Q&A] [answered]
Hi, quick question regarding a group of unavailable sensors. I have 164 entities found on my S40 thermostat, 64 of which are unavailable. They are exclusively sensors and include things like hi/lo pressure switches, coil or outdoor/ambient temp, heating/cooling rates, etc. Not that I would pretend to need them all or even one or two but I am curious what the measurements are and if they would be useful. 

Is there any additional setup needed outside of enabling advanced mode within HA?

Thanks!

**Comment by PeteRager** [comment, marked as answer]
Look for a number entity called diagnostic_level, set it to 2 and those diagnostics should start coming through.
https://github.com/PeteRager/lennoxs30?tab=readme-ov-file#diagnostic_level

When the S40 reboots (like during a power outage), this will get reset to zero.

There is a sample automation that you can use to automatically set it to 2 if this happens. 
https://github.com/PeteRager/lennoxs30/tree/master/samples/diaglevel

**Reply by imcintyr** [reply]
Thank you, Pete. The diag level makes those sensors available. Very helpful.

I do like the idea of the automation to set the diag level after power failures however when I use the code provided I get an error. Specifically:

    _Message malformed: extra keys not allowed @ data['0']_

I have used a couple YAML validation sites which say the code is OK for what it's worth. 

I was having trouble with the code formatting correctly in this message platform so I attached at txt file. Do you see anything that seems off? I compared side by side with yours in github but didn't see any differences in spacing or indentations. 

[Automation Lennox Diag2.txt](https://github.com/PeteRager/lennoxs30/files/14231056/Automation.Lennox.Diag2.txt)

Thanks again!

**Comment by PeteRager** [comment]
Where are you putting that YAML?  You may need to add an automation: to it and then slide everything over 2 spaces.

**Reply by imcintyr** [reply]
Thanks again for your replies. Although I did get the YAML editor to accept the code after some formatting etc, I could not get it to trigger. Since I am not a YAML programmer I turned to the visual editor and created the automation below. Odd thing was it wouldn't trigger either. When I manually ran the automation it changed the diag level to 2 without an issue.

Not sure if this is helpful information but after I replaced the entity_id in the "trigger" section with the entity_id in the "action" section the automation triggers without fail.

I don't know why the visual editor uses two different formats for the entity_id across the automation but I thought it was something to mention.

`alias: New automation
description: ""
trigger:
  - platform: numeric_state
    entity_id:
      - number.harmony_diagnostic_level
    for:
      hours: 0
      minutes: 1
      seconds: 0
    below: 2
condition: []
action:
  - device_id: f48a5dc2e520252cfa89ef730cea0b90
    domain: number
    entity_id: 1a15d012c00b7eed94929a20f5fdf72f
    type: set_value
    value: 2
mode: single
`
